frontend/components/sorting-buttons/component.tsx

```diff
@@ -32,30 +33,50 @@ export const SortingButtons: FC<SortingButtonsProps> = ({
     [options, sortBy]
   );
 
+  const buttonProps = {
+    ...(theme === 'pill' && { size: 'small', theme: 'primary-white' }),
+    ...(theme === 'text' && { size: 'smallest', theme: 'naked' }),
+  } as ButtonProps;
+
   return (
     <div className={className}>
       <div className="flex gap-3">
         <Menu
           Trigger={
             <Button
-              className="whitespace-nowrap"
-              size="small"
-              theme="primary-white"
+              className={cx({
+                'whitespace-nowrap': true,
+                '!px-3 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-green-dark':
+                  theme === 'text',
+              })}
               aria-label={intl.formatMessage({
                 defaultMessage: 'Choose the field to sort by',
                 id: 'DCAMZh',
               })}
               aria-expanded={sortByMenuOpen}
+              {...buttonProps}
             >
               <span className="py-0.5 -mx-2 flex items-center">
-                <span className="text-gray-600">
+                <span className="text-sm text-gray-600">
```

Comment:
On Discover, the font-weight should be `font-medium` and the colour #757575 (`text-gray-700`).